lib-rpc/README.md

```diff
@@ -0,0 +1,86 @@
+# ocamlformat-rpc-lib
+
+The whole API is functorized over an `IO` module defining the blocking interface for reading and writing the data. This module has the following interface:
+
+```ocaml
+  type 'a t (** ['a t] represents a blocking monad state *)
+  type ic (** [ic] represents an input channel *)
+  type oc (** [oc] represents an output channel *)
+
+  val ( >>= ) : 'a t -> ('a -> 'b t) -> 'b t
+  (** [a >>= b] will pass the result of [a] to the [b] function. This is a
+      monadic [bind]. *)
+
+  val return : 'a -> 'a t
+  (** [return a] will construct a constant IO value. *)
+
+  val read : ic -> Csexp.t option t
+  val write : oc -> Csexp.t list -> unit t
+```
+
+A basic implementation of this module can be:
+```ocaml
+module IO = struct
```

Comment:
A flaw with the library is that the module `IO` isn't accessible. Otherwise, this would be better to avoid weird error messages:
```ocaml
module IO : Ocamlformat_rpc_lib.IO = struct
```

Reply:
Well, not really because we need to make `ic` and `oc` escape. That would look too complex for an example.